src/Compilers/Core/Portable/MetadataReader/TypeNameDecoder.cs

```diff
@@ -216,18 +235,36 @@ internal TypeSymbol GetTypeSymbol(MetadataHelpers.AssemblyQualifiedTypeName full
             }
 
             return container;
+
+            // Decode `!n` (or `!!n`) to a type parameter from the current type (or current method).
+            // This is used for decoding extension erasure attributes
+            TypeSymbol tryResolveTypeParameterReference(string serialized)
+            {
+                switch (serialized)
+                {
+                    case ['!', '!', .. var rest] when Int32.TryParse(rest, out int index):
```

Comment:
>.. var rest [](http://example.com/codeflow?start=36&length=11)

We probably should check that the ```rest``` contains only digits to filter out signs and any other weird formats